test/parallel_api/ranges/fill_generate_factory.pass.cpp

```diff
@@ -0,0 +1,49 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include <oneapi/dpl/execution>
+
+#include "support/pstl_test_config.h"
+
+#if _ENABLE_RANGES_TESTING
+#include <oneapi/dpl/ranges>
+#endif
+
+#include "support/utils.h"
+
+#include <iostream>
+
+int32_t
+main()
+{
+#if _ENABLE_RANGES_TESTING
+    constexpr int max_n = 10;
+    auto lambda = [](auto i) { return i * i; };
+
+    using namespace oneapi::dpl::experimental::ranges;
+
+    auto view1 = views::fill(-1, max_n) | views::transform(lambda);
```

Comment:
I guess `views::fill`, `views::generate` and `views::rotate` should be supported on devices too. If so, let's add cases for devices.